Cannot find name 'PromiseWithResolvers'

When trying to build a project that includes @graphql-tools/delegate v10.2.16 as a dependency, I'm getting the following error:

```
node_modules/@graphql-tools/delegate/dist/index.d.ts:214:30 - error TS2304: Cannot find name 'PromiseWithResolvers'.

214 type Deferred<T = unknown> = PromiseWithResolvers<T>;
```

The "PromiseWithResolvers" type is not defined anywhere in the file. I have to set "skipLibCheck" to true in my tsconfig file in order to successfully build.

--- a/.changeset/twenty-fans-share.md
+++ b/.changeset/twenty-fans-share.md
@@ -1,0 +1,5 @@
+---
+'@graphql-tools/delegate': patch
+---
+
+Do not use the global \`PromiseWithResolvers\`

--- a/packages/delegate/src/leftOver.ts
+++ b/packages/delegate/src/leftOver.ts
@@ -3,7 +3,11 @@ import { FieldNode } from 'graphql';
 import { Subschema } from './Subschema.js';
 import { DelegationPlanBuilder, ExternalObject } from './types.js';
 
-export type Deferred<T = unknown> = PromiseWithResolvers<T>;
+export type Deferred<T = unknown> = {
+  promise: Promise<T>;
+  resolve: (value: T | PromiseLike<T>) => void;
+  reject: (reason?: any) => void;
+};
 
 // TODO: Remove this after Node 22
 export { createDeferred };

